--- modules/python/src2/hdr_parser.py	2014-11-05 21:15:45.000000000 +0800
+++ modules/python/src2/hdr_parser.py	2014-11-05 21:15:45.000000000 +0800
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 
 from __future__ import print_function
+from io import open
 import os, sys, re, string
 
 # the list only for debugging. The real list, used in the real OpenCV build, is specified in CMakeLists.txt
@@ -737,7 +738,7 @@
         """
         self.hname = hname
         decls = []
-        f = open(hname, "rt")
+        f = open(hname, "rt", encoding="utf-8")
         linelist = list(f.readlines())
         f.close()
 
